build/commands/lib/config.js

```diff
@@ -52,13 +52,28 @@ var packageConfigBraveCore = function (key) {
   return packageConfig(key, path.join(rootDir, 'src', 'brave'))
 }
 
-const getNPMConfig = (key) => {
+const getNPMConfig = (key, default_value = undefined) => {
   if (!NpmConfig) {
-    const list = run(npmCommand, ['config', 'list', '--json', '--userconfig=' + path.join(rootDir, '.npmrc')])
+    const list = run(npmCommand, ['config', 'list', '--json'], { cwd: rootDir })
     NpmConfig = JSON.parse(list.stdout.toString())
   }
 
-  return NpmConfig[key.join('-').replace(/_/g, '-')] || packageConfigBraveCore(key) || packageConfig(key)
+  // NpmConfig has the multiple copy of the same variable: one from .npmrc
+  // (that we want to) and one from the environment.
+  // https://docs.npmjs.com/cli/v7/using-npm/config#environment-variables
+  const npmConfigValue = NpmConfig[key.join('_')]
```

Comment:
removed the key rewriting of `-` to `_`. AFAIK that was to fix compatibility with different versions of npm. Might not apply anymore if everyone is on node LTS, but we should mention in the commit comments / PR that we're removing compatibility for particular npm version

Reply:
@petemill Are you sure? Maybe you can remember tickets or PRs about this?

This conversion (`some_var` => `some-var` ) happens when you passing [somethings via env vars](https://docs.npmjs.com/cli/v8/using-npm/config#environment-variables) (in fact it actually happens when we call `num run something`, that's why the old code worked)

I've checked docs for npmrc for 6.x, 7.x, 8.x and can't anything about this for regular .npmrc entries.
https://docs.npmjs.com/cli/v6/configuring-npm/npmrc

Reply:
It was for package.json properties which are put in to env variables. So instead of reading from package.json, we read from environment variables.